src/allotropy/allotrope/schema_parser/backup_manager.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Iterator
+from contextlib import contextmanager
+from pathlib import Path
+import shutil
+from typing import Optional, Union
+
+PathType = Union[Path, str]
+
+
+def files_equal(path1: PathType, path2: PathType) -> bool:
+    with open(str(path1)) as file1, open(str(path2)) as file2:
+        for line1, line2 in zip(file1, file2):
```

Comment:
This won't fail if `file1` and `file2` have different numbers of lines but the first (number of lines in smaller file) lines pass your matching function.

The easiest way to fix would be add `strict=True` to `zip`, but that was introduced in 3.10 and we're still supporting 3.9. Seems like the best thing to do in 3.9 is use 
https://docs.python.org/3.3/library/itertools.html#itertools.zip_longest

PEP doc here: https://peps.python.org/pep-0618/

P.S. It appears this was copied from `generate_schemas` -- could you refactor? And since this is a refactor, I'm fine if you want to fix in follow-up.

Reply:
Ah good catch - fixed. 

And whoops, I meant to delete `files_equal` from generate_schemas, done.